store/root.js

```diff
@@ -0,0 +1,88 @@
+import { observable, action, computed } from 'mobx'
+
+import activities from './activities'
+import User from './user'
+import DepositDates from './deposit-dates'
+import Works from './works'
+
+import apiRequest from 'api'
+
+class Root {
+  @observable user = new User()
+
+  @observable dataProvider = null
+
+  @observable activity = null
+
+  @observable statistics = {
+    metadataCount: 0,
+    fullTextCount: 0,
+    doiCount: 0,
+  }
+
+  @observable works = null
+
+  @observable depositDates = null
+
+  @computed
+  get dataProviders() {
+    return this.user.dataProviders
+  }
+
+  @action async init(dataProviderId, activityPath) {
+    await this.user.init()
+    try {
+      this.changeDataProvider(dataProviderId)
+      this.changeActivity(activityPath)
+    } catch (accessError) {
+      const fallbackId = this.user.dataProviders[0].id
+      if (process.env.NODE_ENV !== 'production') {
+        // eslint-disable-next-line no-console
+        console.log(
+          accessError.message,
+          `Fall back to DataProvider ${fallbackId}`
+        )
+      }
+      this.changeDataProvider(fallbackId)
+    }
+  }
+
+  @action changeDataProvider(dataProviderStr) {
+    const dataProviderId = Number.parseInt(dataProviderStr, 10)
+
+    if (this.dataProvider && this.dataProvider.id === dataProviderId) return
+    const dataProvider = this.user.dataProviders.find(
+      ({ id }) => dataProviderId === id
+    )
+
+    if (dataProvider == null) {
+      throw new Error(
+        `User ${this.user.id} does not have access to DataProvider ${dataProviderId}.`
+      )
+    }
+
+    this.dataProvider = dataProvider
+    this.changeActivity('overview')
```

Comment:
IMHO user should stay on the page where they were.

Reply:
No. If data provider changes users starts from home. Secondary navigation does not exist when primary navigates.